tests/test_poisson.py

```diff
@@ -168,30 +142,203 @@ def test_score_predictions():
 
     aicc = uoi_fitter._score_predictions(
         metric='AICc',
         fitter=fitter,
         X=X, y=y, support=support)
-    assert_almost_equal(aicc, aic - 2)
+    assert_allclose(aicc, aic - 2)
 
     bic = uoi_fitter._score_predictions(
         metric='BIC',
         fitter=fitter,
         X=X, y=y, support=support)
-    assert_almost_equal(bic, 2 * ll - np.log(y.size))
+    assert_allclose(bic, 2 * ll - np.log(y.size))
 
     # test invalid metric
     assert_raises(ValueError,
                   uoi_fitter._score_predictions,
                   'fake',
                   fitter,
                   X, y, support)
+
+
+def test_poisson_intercept_fitter_no_features():
+    """Tests the PoissonInterceptFitterNoFeatures class."""
+    y = np.array([0, 1, 2])
+    poisson = PoissonInterceptFitterNoFeatures(y)
+
+    X = np.random.normal(size=(10, y.size))
+    assert_equal(poisson.intercept_, 0.)
+    assert_equal(poisson.predict(X), 1.)
+    assert_equal(poisson.predict_mean(X), 1.)
+
+
+def test_poisson_loss_and_grad():
+    """Tests the poisson loss and gradient function."""
+    n_features = 5
+    n_samples = 100
+
+    # test with empty coefficients
+    X = np.random.normal(loc=0, scale=1., size=(n_samples, n_features))
+    y = np.random.normal(loc=0, scale=1, size=(n_samples))
+
+    loss, grad = _poisson_loss_and_grad(np.zeros(n_features), X, y, 0.)
+
+    assert_equal(loss, 1.)
+    assert_allclose(grad, -np.dot(X.T, y - 1) / n_samples)
+
+    # test with non-empty coefficients
+    X = np.ones((n_samples, n_features))
+    y = np.zeros(n_samples)
+
+    loss, grad = _poisson_loss_and_grad(np.ones(n_features), X, y, 0)
+    assert_equal(loss, np.exp(n_features).mean())
+    assert_allclose(grad, np.mean(X * np.exp(n_features), axis=0))
+
+
+def test_poisson_no_intercept():
+    """Tests the Poisson fitter with no intercept."""
+    n_features = 3
+    n_samples = 10000
+
+    # create data
+    X = np.random.normal(loc=0, scale=1. / 8, size=(n_samples, n_features))
+    beta = np.array([0.5, 1.0, 1.5])
+    eta = np.dot(X, beta)
+    y = np.random.poisson(np.exp(eta))
+
+    # lbfgs
+    poisson = Poisson(alpha=0., l1_ratio=0., fit_intercept=False,
+                      solver='lbfgs', max_iter=5000)
+    poisson.fit(X, y)
+    assert_allclose(poisson.coef_, beta, rtol=0.5)
+
+    # coordinate descent
+    poisson = Poisson(alpha=0., l1_ratio=0., fit_intercept=False, solver='cd',
+                      max_iter=5000)
+    poisson.fit(X, y)
+    assert_allclose(poisson.coef_, beta, rtol=0.5)
+
+    # broken solver
+    poisson = Poisson(alpha=0., l1_ratio=0., fit_intercept=False, solver='ruff',
+                      max_iter=5000)
+    assert_raises(ValueError, poisson.fit, X, y)
+
+
+def test_poisson_warm_start():
+    """Tests if the warm start initializes coefficients correctly."""
+    n_features = 3
+    n_samples = 10000
+
+    # create data
+    X = np.random.normal(loc=0, scale=1. / 8, size=(n_samples, n_features))
```

Comment:
It might be worthwhile to make a make_poisson_regression function in ``utils.py`` since this gets used in a few tests.